### PR TITLE
Avoid rendering unknown fields and empty layouts

### PR DESCRIPTION
**Pull request by AhmetTanakol**
Return null if an unknown field is rendered
Set the layout as null if the type of field is object with empty properties when generating uischema

### CODE REVIEW

File: packages/editor/src/helpers/NonEmptyLayout.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  RendererProps,
+  Tester,
+} from '@jsonforms/core';
+import { connectToJsonForms } from '@jsonforms/react';
+import * as _ from 'lodash';
+
+/**
+ * Checks whether the given UI schema contains only label field or not.
+ */
+export const isInstanceOfLayout: Tester =
```
Comment (edgarmueller) [inline packages/editor/src/helpers/NonEmptyLayout.tsx]: Rename tester to `isEmptyLayout`

File: packages/editor/src/helpers/NonEmptyLayout.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+export const nonEmptyLayoutTester: RankedTester =
+  rankWith(3, isInstanceOfLayout);
+
+/**
+ * NonEmptyLayoutRenderer is a custom layout renderer to prevent rendering empty layouts
+ *
+ * @returns null
+ * @constructor
+ */
+export const NonEmptyLayoutRenderer = (
```
Comment (edgarmueller) [inline packages/editor/src/helpers/NonEmptyLayout.tsx]: Add test case for `NonEmptyLayoutRenderer`.

File: packages/material/src/controls/MaterialInputControl.tsx [hunk 3]
```diff
@@ -61,23 +63,27 @@ export class MaterialInputControl extends Control<ControlProps, ControlState> {
...
-          {computeLabel(isPlainLabel(label) ? label : label.default, required)}
-        </InputLabel>
-        <DispatchField uischema={uischema} schema={schema} path={parentPath} />
-        <FormHelperText error={!isValid}>
-          {!isValid ? formatErrorMessage(errors) : showDescription ? description : null}
-        </FormHelperText>
-      </FormControl>
-    );
+    const field = _.maxBy(fields, r => r.tester(uischema, schema));
+    if (field === undefined || field.tester(uischema, schema) === -1) {
```
Comment (edgarmueller) [inline packages/material/src/controls/MaterialInputControl.tsx]: A couple of things: 
1. Use `NOT_APPLICABLE` constant.
2. Are there any other controls that need to be updated to make the behavior consistent, e.g. `MaterialBooleanControl`? What about the vanilla renderer set? Ideally, this check could be moved into a [HOC](https://reactjs.org/docs/higher-order-components.html)
3. Add test case(s)

**Comment by AhmetTanakol** [comment]
Moving this to jsonforms-ui-editor

**Pull request closed without merging**